### PR TITLE
[NEW model] early warning notification by Andreas & Timothy patch 1

### LINKED ISSUE #421: [New Model]: early warning notification
## Model Description
The new model enables the notification function about findings from joint field monitoring between two participants fo the supply chain. This functionality is part of the early warning feature from the use case quality.

## Relevant Standards
It belongs to the same group of aspects as the MessageHeader Aspect Model: io.catenax.shared.message_header
 
## Example Data
{
  "header": {
    "senderBpn": "BPNL7588787849VQ",
    "relatedMessageId": "d9452f24-3bf3-4134-b3eb-68858f1b2362",
    "expectedResponseBy": "2023-06-19T21:24:00+07:00",
    "context": "urn:samm:io.catenax.<ASPECT-MODEL-NAME>:1.x.x",
    "messageId": "3b4edc05-e214-47a1-b0c2-1d831cdd9ba9",
    "receiverBpn": "BPNL6666787765VQ",
    "sentDateTime": "2023-06-19T21:24:00+07:00",
    "version": "urn:samm:io.catenax.message_header:1.0.0"
  },
  "content": {
    "qualityAppFilter": {
      "uniqueIdentifiers": [
        {
          "serialPartNumber": "436347347.4343884384.FTG.538348",
          "anonymizedVIN": "3747429FGH382923974682",
          "catenaXId": "urn:uuid:c227a880-b82b-40f7-846c-3942ddf26c29"
        }
      ],
      "populationFilter": [
        {
          "aspectModel": "urn:bamm:io.catenax.manufactured_parts_quality_information:1.0.0",
          "filters": [
            {
              "from": "2019-02-01T00:00:00",
              "to": "2019-07-30T23:59:59",
              "aspectProperty": "production.productionDate",
              "dataType": "daterange",
              "valueList": [
                {
                  "value": "123-0.740-3434-A"
                }
              ]
            }
          ]
        }
      ]
    },
    "earlyWarningAttachmentLink": "subprotocolBody: catalogEndpoint=https://test.com/catalog;dataset=79afc338-f7ea-4255-a17d-ba6faf40d2b5",
    "anomalyDescription": "There is an unusual number of DTCs occuring in the population described by the property qualityAppFiilter",
    "qualityTaskID": "BPN-811_2022_000001"
  }
}

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by Timothy-Ra**
## Description
The new model enables the notification function about findings from joint field monitoring between two participants of the supply chain. This functionality is part of the early warning feature from the use case quality. 

Closes #421 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by agg3fe** [comment]
@Timothy-Ra Did you create any Github issue for this Pull Request? If not, then please first create an issue [here](https://github.com/eclipse-tractusx/sldt-semantic-models/issues).After that we go for approval process.

File: io.catenax.early_warning_notification/1.0.0/metadata.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/metadata.json:1]: This file should only contain `{ "status" : "release"} `. please delete the content of this file.

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 2]
```diff
@@ -0,0 +1,153 @@
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:org.eclipse.examples:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:8]: please replace the `examples` with `early_warning_notification`

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+   samm:exampleValue "subprotocolBody: catalogEndpoint=https://test.com/catalog;dataset=79afc338-f7ea-4255-a17d-ba6faf40d2b5" .
+
+:QualityAppFilterCharacteristic a samm:Characteristic ;
+   samm:preferredName "Quality App Filter Characteristic"@en ;
+   samm:description "Characteristic for the quality app filter"@en ;
+   samm:dataType :QualityAppFilter .
+
+:QualityAppFilter a samm:Entity ;
+   samm:preferredName "Affected Population FIlter"@en ;
+   samm:description "A set of filters and/or a list definieng the affected population"@en ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:62]: typo `definieng`

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+   samm:properties ( :populationFilter :uniqueIdentifiers ) .
+
+:populationFilter a samm:Property ;
+   samm:preferredName "Population Filter"@en ;
+   samm:description "Filters on the common early warning data set to specify a population"@en ;
+   samm:characteristic :PopulationFilterCharateristic .
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:68]: typo `Charateristic`

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+:uniqueIdentifiers a samm:Property ;
+   samm:preferredName "listOfSerialNumbers"@en ;
+   samm:description "List of unique identifiers to explicitly specifiy parts or vehicles exhibiting the issue described in the property \"anomalyDescription\""@en ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:72]: typo `specifiy`

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+   samm:characteristic :UniqueIdentifiersCharacteristic .
+
+:PopulationFilterCharateristic a samm-c:List ;
+   samm:preferredName "Population Filter Charateristic"@en ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:76]: typo `Charateristic`

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+:serialPartNumber a samm:Property ;
+   samm:preferredName "Serial Part Number"@en ;
+   samm:description "A unique serial number for a produced part"@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "436347347.4343884384.FTG.538348" .
+
+:ValuesCharacteristic a samm-c:List ;
+   samm:preferredName "Values Characteristic"@en ;
+   samm:description "Caracteristic for the list of values"@en ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:151]: typo `Caracteristic`

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+:anomalyDescription a samm:Property ;
+   samm:preferredName "Anomaly Description"@en ;
+   samm:description "Some text describing the detected anomaly in the data"@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "There is an unusual number of DTCs occuring in the population described by the property qualityAppFiilter" .
+
+:qualityAppFilter a samm:Property ;
+   samm:preferredName "Quality App Filter"@en ;
+   samm:description "A filter to define the population of affected parts in a C-atnea-X quality app"@en ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:46]: is `C-atnea-X` right?

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+   samm:description "A set of filters and/or a list definieng the affected population"@en ;
+   samm:properties ( :populationFilter :uniqueIdentifiers ) .
+
+:populationFilter a samm:Property ;
+   samm:preferredName "Population Filter"@en ;
+   samm:description "Filters on the common early warning data set to specify a population"@en ;
+   samm:characteristic :PopulationFilterCharateristic .
+
+:uniqueIdentifiers a samm:Property ;
+   samm:preferredName "listOfSerialNumbers"@en ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:71]: `preferredName` should be plain english and not camel case.

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+   samm:preferredName "VAN"@en ;
+   samm:description "A unique but anonymized identifier for a vehicle"@en ;
+   samm:see <urn:samm:io.catenax.vehicle.product_description:3.0.0> ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "3747429FGH382923974682" .
+
+:catenaXId a samm:Property ;
+   samm:preferredName "Catena-X ID"@en ;
+   samm:description "A unique identifier for a part or a vehicle. The identifier relates to a Catena-X digital twin"@en ;
+   samm:characteristic samm-c:Text ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:140]: you can make use of shared aspect model (io.catenax.shared.uuid:1.0.0) for UUID.
you can refer this https://github.com/eclipse-tractusx/sldt-semantic-models/blob/a07bdd5b1515767d4f8d07deb47cf2a4aa728bef/io.catenax.quality_message_content/1.0.0/QualityMessageContent.ttl#L35C9-L35C17

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 11]
```diff
@@ -0,0 +1,153 @@
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:1]: please add license headers also.

**Review by agg3fe — CHANGES_REQUESTED** [review]
Please check my comments

**Comment by Timothy-Ra** [comment]
> Please check my comments

@agg3fe we changed the model according to your remarks and made some changes due to new infos on our side. Please check once again. Thank you :)

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl (hunk 10 above)
Comment (agg3fe) [reply io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:146]: @Timothy-Ra please check this also.

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.early_warning_notification:1.0.0#> .
+@prefix ext-header: <urn:samm:io.catenax.shared.message_header:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:30]: you can consider reference to new message header 2.0.0 as new version is released now.

**Comment by agg3fe** [comment]
@Timothy-Ra please check my comments

**Comment by Timothy-Ra** [comment]
> @Timothy-Ra please check my comments

@andreas-hutter and I updated according to your comments and some last minute changes from our side.

@agg3fe please validade

File: io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+   samm:properties ( :aspectProperty :aspectModel [ samm:property :rangeFrom; samm:optional true ] [ samm:property :rangeTo; samm:optional true ] :dataType [ samm:property :values; samm:optional true ] ) .
+
+:anonymizedVIN a samm:Property ;
+   samm:preferredName "VAN"@en ;
+   samm:description "A unique but anonymized identifier for a vehicle"@en ;
+   samm:see <urn:samm:io.catenax.vehicle.product_description:3.0.0> ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "3747429FGH382923974682" .
+
+:catenaXId a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:122]: if this is a unique catenaXId,  you can make use of shared aspect model Uuid.
example: https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.asset_tracker_links/2.0.0/AssetTrackerLinks.ttl#L38

Comment (Timothy-Ra) [reply io.catenax.early_warning_notification/1.0.0/EarlyWarningNotification.ttl:122]: okay, i misunderstood, I changed it

**Review by agg3fe — APPROVED** [review]
LGTM

**Review by agg3fe — CHANGES_REQUESTED** [review]
please add dependency

**Comment by agg3fe** [comment]
Hi @Timothy-Ra, As I suggested above. please add the dependency which is missing. your model is invalid right now.

**Comment by andreas-hutter** [comment]
I have added the dependecy as requested (and removed 3 no longer needed properties). Can you please check the model validity again ?

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by agg3fe** [comment]
@Timothy-Ra please check eclipsefdn/eca step above https://api.eclipse.org/git/eca/status/gh/eclipse-tractusx/sldt-semantic-models/484. 
It is giving error.

File: io.catenax.early_warning_notification/RELEASE_NOTES.md [hunk 14]
```diff
@@ -0,0 +1,11 @@
+Changelog
+All notable changes to this model will be documented in this file.
+
+[Unreleased]
+[1.0.0] - 2023-12-11
```
Comment (agg3fe) [inline io.catenax.early_warning_notification/RELEASE_NOTES.md:5]: please change it to today's date if you need MS3 today.

**Comment by Timothy-Ra** [comment]
thanks so much @agg3fe and @andreas-hutter for the great cooperation and that we finished that model :)

**Pull request merged**